internal/core/entities/allocation/allocation.go

```diff
@@ -0,0 +1,40 @@
+// MIT License
+//
+// Copyright (c) 2021 TFG Co
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package allocation
+
+// MatchAllocation is the struct that defines the configuration for the match
+// allocation in the Game Rooms.
+type MatchAllocation struct {
+	// MaxMatches defined the maximum number of matches that a room can host.
+	MaxMatches int `validate:"required,min=1,max=30"`
+	// MinFreeSlots defines the minimum number of free slots that a room must
+	// have to become available for allocation.
+	MinFreeSlots int `validate:"required,min=1"`
```

Comment:
we should add `max=30` here because the upper limit is MaxMatches, right? And in addition we'll validate one value against the other as well but `max=30` we "gain" a validation prior to us checking one against the other